### PR TITLE
Fix sig n cipher functions

### PR DESCRIPTION
**Pull request by ToddyTheNoobDud**
this fixes everything from the sig related stuff (example: Could not parse n transform function, 403, and others)

### CODE REVIEW

**Comment by ToddyTheNoobDud** [comment]
i forgot to remove my own performance rewrite (💀 ), only needs the [a800d42](https://github.com/distubejs/ytdl-core/pull/139/commits/a800d422a52c53c512443e31d8f75b6b2333b899) commit to work again...

File: lib/agent.js [hunk 1]
```diff
@@ -2,39 +2,45 @@ const { ProxyAgent } = require("undici");
...
-      return "strict";
+      return "Strict";
     case "lax":
-      return "lax";
+      return "Lax";
     case "no_restriction":
     case "unspecified":
     default:
-      return "none";
+      return "None";
```
Comment (skick1234) [inline lib/agent.js:14]: I don't think this is needed

File: lib/agent.js [hunk 2]
```diff
@@ -2,39 +2,45 @@ const { ProxyAgent } = require("undici");
...
+    value: cookie.value,
+    expires: typeof cookie.expirationDate === "number" ? new Date(cookie.expirationDate * 1000) : "Infinity",
+    domain: canonicalDomain(cookie.domain),
+    path: cookie.path || '/',
+    secure: cookie.secure || false,
+    httpOnly: cookie.httpOnly || false,
+    sameSite: convertSameSite(cookie.sameSite),
+    hostOnly: cookie.hostOnly || false,
+  });
+};
```
Comment (skick1234) [inline lib/agent.js:33]: The inline won't allocate a new object basically because it doesn't run

File: lib/agent.js [hunk 3]
```diff
@@ -47,26 +53,24 @@ const addCookies = (exports.addCookies = (jar, cookies) => {
...
-  );
+
+  const cookieArray = cookies.split(";").map(c => Cookie.parse(c)).filter(Boolean);
+  addCookies(jar, cookieArray);
 };
 
-const createAgent = (exports.createAgent = (cookies = [], opts = {}) => {
-  const options = Object.assign({}, opts);
+const createAgent = (cookies = [], opts = {}) => {
+  const options = { ...opts };
```
Comment (skick1234) [inline lib/agent.js:73]: Object.assign is faster than spread

File: lib/agent.js [hunk 4]
```diff
@@ -77,24 +81,30 @@ const createAgent = (exports.createAgent = (cookies = [], opts = {}) => {
...
+// Default agent
+const defaultAgent = createAgent();
+
+module.exports = {
+  addCookies,
+  addCookiesFromString,
+  createAgent,
+  createProxyAgent,
+  defaultAgent,
+};
```
Comment (skick1234) [inline lib/agent.js:110]: Not necessary. But I will apply this if someone create a pull request contains this.

File: lib/sig.js [hunk 5]
```diff
@@ -19,93 +18,89 @@ exports.getFunctions = (html5playerfile, options) =>
...
+    `\\(${SCVR}=String\\.fromCharCode\\(110\\),${SCVR}=${SCVR}\\.get\\(${SCVR}\\)\\)&&\\(${SCVR}=(${MCR})(?:${AAR})?\\(${
+      SCVR
+    }\\)`,
+    `\\.get\\("n"\\)\\)&&\\(${SCVR}=(${MCR})(?:${AAR})?\\(${SCVR}\\)`,
+    // Skick regexps
+    '(\\w+).length\\|\\|\\w+\\(""\\)',
+    '\\w+.length\\|\\|(\\w+)\\(""\\)',
+  ];
+  
+const timestampPattern = /(?:signatureTimestamp|sts):(\d+)/;
```
Comment (skick1234) [inline lib/sig.js:77]: timestampPattern is not used

File: lib/agent.js [hunk 6]
```diff
@@ -77,24 +81,30 @@ const createAgent = (exports.createAgent = (cookies = [], opts = {}) => {
...
-      },
-    },
-    options,
-  );
+  
+  const proxyOptions = {
+    factory: (origin, opts) => new CookieClient(origin, { ...opts, cookies: { jar } }),
+    ...options,
+  };
+
```
Comment (skick1234) [inline lib/agent.js:97]: Object.assign is faster than spread

File: lib/sig.js [hunk 7]
```diff
@@ -19,93 +18,89 @@ exports.getFunctions = (html5playerfile, options) =>
...
 ];
-
-// LavaPlayer regexps
 const VARIABLE_PART = "[a-zA-Z_\\$][a-zA-Z_0-9]*";
-const VARIABLE_PART_DEFINE = `\\"?${VARIABLE_PART}\\"?`;
-const BEFORE_ACCESS = '(?:\\[\\"|\\.)';
-const AFTER_ACCESS = '(?:\\"\\]|)';
+const VARIABLE_PART_DEFINE = `\\"?(${VARIABLE_PART})\\"?`;
+const BEFORE_ACCESS = "(?:\\[\\\"|\\.)";
+const AFTER_ACCESS = "(?:\\\"\\]|)";
```
Comment (skick1234) [inline lib/sig.js:33]: Unnecessary changes

File: lib/sig.js [hunk 8]
```diff
@@ -202,17 +194,13 @@ const extractNTransformWithName = body => {
...
   const nTransformFunc = getExtractFunctions([extractNTransformFunc, extractNTransformWithName], body);
   if (!nTransformFunc && !nTransformWarning) {
-    // This is optional, so we can continue if it's not found, but it will bottleneck the download.
     console.warn(
       "\x1b[33mWARNING:\x1B[0m Could not parse n transform function.\n" +
-        `Please report this issue with the "${utils.saveDebugFile(
-          "base.js",
-          body,
-        )}" file on https://github.com/distubejs/ytdl-core/issues.`,
+      `https://github.com/ToddyTheNoobDud/ytdl-core-toddys/issues`,
```
Comment (skick1234) [inline lib/sig.js:203]: Revert this

**Review by skick1234 — CHANGES_REQUESTED** [review]
I will close this and fix the regex myself.
You can split this into some pull requests for easier merging.

**Pull request closed without merging**